sale_rental/models/sale_rental.py

```diff
@@ -0,0 +1,163 @@
+                if rental.sell_order_line_ids and\
+                        rental.sell_order_line_ids[0].move_ids:
+                    sell_move = rental.sell_order_line_ids[0].move_ids[-1]
+                state = 'ordered'
+                if out_move and out_move.state == 'done':
+                    state = 'out'
+                    if in_move:
+                        if in_move.state == 'done':
+                            state = 'in'
+                        elif in_move.state == 'cancel' and sell_move:
+                            state = 'sell_progress'
+                            if sell_move.state == 'done':
+                                state = 'sold'
+                    elif sell_move:
+                        state = 'sell_progress'
+                        if sell_move.state == 'done':
+                            state = 'sold'
+                        elif sell_move.state == 'cancel':
+                            state = 'out'
+                if rental.start_order_line_id.state == 'cancel':
+                    state = 'cancel'
+            rental.in_move_id = in_move
+            rental.out_move_id = out_move
+            rental.state = state
+            rental.sell_move_id = sell_move
+
+    @api.depends(
+        'extension_order_line_ids.end_date',
+        'extension_order_line_ids.state',
+        'start_order_line_id.end_date')
+    def _compute_end_date(self):
+        for rental in self:
+            end_date = False
+            if rental.start_order_line_id:
+                end_date = rental.start_order_line_id.end_date
+
+            for extension in rental.extension_order_line_ids:
+                if (
+                        extension.state in ('sale', 'done') and
+                        end_date and
+                        extension.end_date and
+                        extension.end_date > end_date):
+                    end_date = extension.end_date
+            rental.end_date = end_date
+
+    start_order_line_id = fields.Many2one(
+        'sale.order.line', string='Rental SO Line', readonly=True)
+    start_date = fields.Date(
+        related='start_order_line_id.start_date', readonly=True, store=True)
+    rental_product_id = fields.Many2one(
+        'product.product', related='start_order_line_id.product_id',
+        string="Rental Service", readonly=True, store=True)
+    rented_product_id = fields.Many2one(
+        'product.product',
+        related='start_order_line_id.product_id.rented_product_id',
+        string="Rented Product", readonly=True, store=True)
+    rental_qty = fields.Float(
+        related='start_order_line_id.rental_qty', readonly=True, store=True)
+    start_order_id = fields.Many2one(
+        'sale.order', related='start_order_line_id.order_id',
+        string='Rental SO', readonly=True, store=True)
+    company_id = fields.Many2one(
+        'res.company', related='start_order_line_id.order_id.company_id',
```

Comment:
```suggestion
        'res.company', related='start_order_line_id.company_id',
```
See https://github.com/odoo/odoo/blob/12.0/addons/sale/models/sale.py#L1216